### PR TITLE
Update to5.16

### PR DESCRIPTION
**Pull request by craffit**
For more info see https://jira.stackstate.io/browse/STAC-1276

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ sudo: false
 branches:
   only:
     - master
-    - serviceDiscoveryTest
+    - updateTo5.16
```
Comment (whazenberg) [inline .travis.yml:6]: let's remove this when everything is done

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 This document aims to provide some basic guidelines to contribute to this repository, but keep in mind that these are just guidelines, not rules; use your best judgment and feel free to propose changes to this document in a pull request.
 
+Want help with your PRs on this project? We offer [office hours](https://github.com/DataDog/dd-agent/wiki/Community-Office-Hours) twice a month via Slack and Google Hangouts.
```
Comment (whazenberg) [inline CONTRIBUTING.md]: hmmm. do we want to remove this?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -26,6 +28,11 @@ Your Pull Request **must** always pass the Travis/Appveyor tests before being me
 
 _If you are adding a dependency (python module, library, ...), please check the [corresponding section](#add-dependencies)._
 
+## [Integrations](https://github.com/DataDog/integrations-core)
```
Comment (whazenberg) [inline CONTRIBUTING.md]: maybe put our sts-agent-integrations-core repo here?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -26,6 +28,11 @@ Your Pull Request **must** always pass the Travis/Appveyor tests before being me
...
+
+All checks have been moved to the [Integration SDK](https://github.com/DataDog/integrations-core). Please look there to submit related issues, PRs, or review the latest changes.
```
Comment (whazenberg) [inline CONTRIBUTING.md]: our repo?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -26,6 +28,11 @@ Your Pull Request **must** always pass the Travis/Appveyor tests before being me
...
+
+For new integrations, please open a pull request in the [integrations extras repo](https://github.com/DataDog/integrations-extras)
```
Comment (whazenberg) [inline CONTRIBUTING.md]: our repo?

File: README.md [hunk 6]
```diff
@@ -6,6 +6,10 @@ The StackState Agent collects events and metrics and brings them to your
 It includes telemetry information from various sources, as well as topology
 information based on [connbeat](https://github.com/raboof/connbeat).
 
+# [Integration SDK](https://github.com/DataDog/integrations-core)
```
Comment (whazenberg) [inline README.md]: our repo?

File: README.md [hunk 7]
```diff
@@ -6,6 +6,10 @@ The StackState Agent collects events and metrics and brings them to your
...
+
+All checks have been moved to the [Integration SDK](https://github.com/DataDog/integrations-core). Please look there to submit related issues, PRs, or review the latest changes.
```
Comment (whazenberg) [inline README.md]: our repo?

File: README.md (hunk 6 above)
Comment (whazenberg) [reply README.md]: when it is not shipped we can opt to not change it.

File: Rakefile [hunk 8]
```diff
@@ -48,11 +18,15 @@ CLOBBER.include '**/*.pyc'
...
-  ENV['INTEGRATIONS_DIR'] = File.join(rakefile_dir, 'embedded')
+  # Commented out to make 'rake run' work with integrations in ../integrations and
+  # configuration in ./confi.d/*.yaml
+  # ENV['INTEGRATIONS_DIR'] = File.join(rakefile_dir, 'embedded')
+  ENV['CHECKSD_OVERRIDE'] = File.join(rakefile_dir, 'tests/checks/fixtures/checks')
   ENV['PIP_CACHE'] = File.join(rakefile_dir, '.cache/pip')
   ENV['VOLATILE_DIR'] = '/tmp/dd-agent-testing'
   ENV['CONCURRENCY'] = ENV['CONCURRENCY'] || '2'
   ENV['NOSE_FILTER'] = 'not windows'
+  ENV['JMXFETCH_URL'] = 'https://s3-eu-west-1.amazonaws.com/sts-jmxfetch'
```
Comment (whazenberg) [inline Rakefile:29]: magic rename or legit?
Is it this one: https://github.com/StackVista/jmxfetch ?

File: checks/prometheus_check.py [hunk 9]
```diff
@@ -0,0 +1,357 @@
+# (C) Datadog, Inc. 2017
```
Comment (whazenberg) [inline checks/prometheus_check.py]: maybe we want to edit the header information, at least do it consistently

File: config.py [hunk 10]
```diff
@@ -19,26 +19,36 @@
...
 )
+from utils.windows_configuration import get_registry_conf, get_windows_sdk_check
+
 
 # CONSTANTS
-AGENT_VERSION = "1.0.0"
 STACKSTATE_CONF = "stackstate.conf"
 UNIX_CONFIG_PATH = '/etc/sts-agent'
 MAC_CONFIG_PATH = '/opt/stackstate-agent/etc'
+AGENT_VERSION = "5.16.0"
```
Comment (whazenberg) [inline config.py]: Do we have to change this to our version number?

File: config.py [hunk 11]
```diff
@@ -1103,16 +1215,20 @@ def get_logging_config(cfg_path=None):
...
+        logging_config['forwarder_log_file'] = os.path.join(_windows_commondata_path(), 'StackState', 'logs', 'forwarder.log')
+        logging_config['dogstatsd_log_file'] = os.path.join(_windows_commondata_path(), 'StackState', 'logs', 'dogstatsd.log')
         logging_config['jmxfetch_log_file'] = os.path.join(_windows_commondata_path(), 'StackState', 'logs', 'jmxfetch.log')
+        logging_config['service_log_file'] = os.path.join(_windows_commondata_path(), 'StackState', 'logs', 'service.log')
+        logging_config['log_to_syslog'] = False
     else:
         logging_config['collector_log_file'] = '/var/log/stackstate/collector.log'
         logging_config['forwarder_log_file'] = '/var/log/stackstate/forwarder.log'
-        logging_config['dogstatsd_log_file'] = '/var/log/stackstate/stsstatsd.log'
+        logging_config['dogstatsd_log_file'] = '/var/log/stackstate/dogstatsd.log'
```
Comment (whazenberg) [inline config.py]: maybe we need to change dogstatsd to stsstatsd.

File: emitter.py [hunk 12]
```diff
@@ -129,7 +120,60 @@ def http_emitter(message, log, agentConfig, endpoint):
...
+    legacy_url = "{0}/intake/{1}?api_key={2}".format(agentConfig['dd_url'], endpoint, api_key)
+    metrics_endpoint = "{0}/api/v1/series?api_key={1}".format(agentConfig['dd_url'], api_key)
+
+    legacy_payload, metrics_payload = split_payload(message)
+
+    # Post legacy payload
+    post_payload(legacy_url, legacy_payload, agentConfig, log)
+
+    # Post metrics payload
+    post_payload(metrics_endpoint, metrics_payload, agentConfig, log)
```
Comment (whazenberg) [inline emitter.py:173]: we probably have to check whether `simulator record -t <file>` still works

Comment (whazenberg) [reply emitter.py:173]: I'm not expecting any problems, I guess I'm just putting stickies here ;)

File: packaging/datadog-agent/source/setup_agent.sh [hunk 13]
```diff
@@ -11,7 +11,7 @@ set -u
 # SCRIPT KNOBS
 #######################################################################
 # Update for new releases, will pull this tag in the repo
-DEFAULT_AGENT_VERSION="5.9.1"
+DEFAULT_AGENT_VERSION="5.14.1"
```
Comment (whazenberg) [inline packaging/datadog-agent/source/setup_agent.sh:14]: our version?

File: checks/prometheus_check.py (hunk 9 above)
Comment (whazenberg) [reply checks/prometheus_check.py]: please check other files as well.

**Review by whazenberg — CHANGES_REQUESTED** [review]

**Comment by whazenberg** [comment]
Do we have a build yet that I can manually test?

File: CONTRIBUTING.md (hunk 2 above)
Comment (craffit) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 3 above)
Comment (craffit) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 4 above)
Comment (craffit) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 5 above)
Comment (craffit) [reply CONTRIBUTING.md]: done

File: README.md (hunk 6 above)
Comment (craffit) [reply README.md]: done

File: README.md (hunk 7 above)
Comment (craffit) [reply README.md]: done

File: Rakefile (hunk 8 above)
Comment (craffit) [reply Rakefile:29]: legit

File: checks/prometheus_check.py (hunk 9 above)
Comment (craffit) [reply checks/prometheus_check.py]: done

File: config.py (hunk 10 above)
Comment (craffit) [reply config.py]: Made this a bogus number to not confuse the user

File: config.py (hunk 11 above)
Comment (craffit) [reply config.py]: done

File: emitter.py (hunk 12 above)
Comment (craffit) [reply emitter.py:173]: done that

File: packaging/datadog-agent/source/setup_agent.sh (hunk 13 above)
Comment (craffit) [reply packaging/datadog-agent/source/setup_agent.sh:14]: irrelevant code

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -135,11 +123,11 @@ And replace, `my_module` with your module name, `default_version` by the version
 
 And it's done for `omnibus-software`!
 
-#### dd-agent-omnibus
+#### sts-agent-omnibus
 
 First, fork `omnibus-software`, and create your branch.
 
-Then add `dependency "my_module"` to `config/projects/datadog-agent.rb`: (please respect alphabetical sort)
+Then add `dependency "my_module"` to `config/projects/stckstate-agent.rb`: (please respect alphabetical sort)
```
Comment (whazenberg) [inline CONTRIBUTING.md]: stck -> stack*

**Review by whazenberg — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 14 above)
Comment (craffit) [reply CONTRIBUTING.md]: done

**Review by whazenberg — APPROVED** [review]

**Pull request merged**